src/app/dataset-view/additional-components/data-component/query-and-result-sections/query-and-result-sections.component.ts

```diff
@@ -0,0 +1,169 @@
+import { DataRow, DatasetRequestBySql } from "src/app/interface/dataset.interface";
+import ProjectLinks from "src/app/project-links";
+import { QueryExplainerService } from "src/app/query-explainer/query-explainer.service";
+import { QueryExplainerProofResponse } from "src/app/query-explainer/query-explainer.types";
+import { FileUploadService } from "src/app/services/file-upload.service";
+import { NavigationService } from "src/app/services/navigation.service";
+import { Clipboard } from "@angular/cdk/clipboard";
+import { AppConfigService } from "src/app/app-config.service";
+import { DatasetSubscriptionsService } from "src/app/dataset-view/dataset.subscriptions.service";
+
+@Component({
+    selector: "app-query-and-result-sections",
+    templateUrl: "./query-and-result-sections.component.html",
+    styleUrls: ["./query-and-result-sections.component.scss"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class QueryAndResultSectionsComponent extends BaseComponent implements OnInit {
+    @Input({ required: true }) public sqlLoading: boolean;
+    sqlErrorMarker$: Observable<string>;
+    public dataUpdate$: Observable<MaybeNull<DataUpdate>>;
+    @Input({ required: true }) public sqlRequestCode: string;
+    public currentData: DataRow[] = [];
+    public isAllDataLoaded = false;
+    @Output() public runSQLRequestEmit = new EventEmitter<DatasetRequestBySql>();
+
+    private loggedUserService = inject(LoggedUserService);
+    private queryExplainerService = inject(QueryExplainerService);
+    private fileUploadService = inject(FileUploadService);
+    private navigationService = inject(NavigationService);
+    private clipboard = inject(Clipboard);
+    private toastService = inject(ToastrService);
+    private appConfigService = inject(AppConfigService);
+    private cdr = inject(ChangeDetectorRef);
+    private datasetSubsService = inject(DatasetSubscriptionsService);
+
+    public ngOnInit(): void {
+        this.datasetSubsService.emitSqlQueryDataChanged(null);
+        this.sqlErrorMarker$ = this.datasetSubsService.sqlErrorOccurrences.pipe(
+            map((data: DataSqlErrorUpdate) => data.error),
+        );
+        this.dataUpdate$ = this.datasetSubsService.sqlQueryDataChanges.pipe(
+            tap((dataUpdate: MaybeNull<DataUpdate>) => {
+                if (dataUpdate) {
+                    if (dataUpdate.currentVocab?.offsetColumn) {
+                        this.offsetColumnName = dataUpdate.currentVocab.offsetColumn;
+                    }
+                    this.isAllDataLoaded = dataUpdate.content.length < this.rowsLimit;
+                    this.currentData = this.skipRows
+                        ? [...this.currentData, ...dataUpdate.content]
+                        : dataUpdate.content;
+                    this.datasetSubsService.resetSqlError();
+                }
+            }),
+        );
```

Comment:
This is now a child presentation-level component. It should receive data from the top component and be dumb. Presentation components should never access services directly, but should receive data from ports and emit events from clicks.